--- ./src/Makefile.am.org	Sun Aug 26 14:49:35 2001
+++ ./src/Makefile.am	Wed Jul 10 17:46:21 2002
@@ -4,7 +4,9 @@
 
 INCLUDES = \
 	-I$(top_srcdir)/intl \
-	$(GNOME_INCLUDEDIR)
+	$(GNOME_INCLUDEDIR) \
+	$(GDKPIXBUFF_CFLAGS) \
+	$(LIBGLADE_CFLAGS)
 
 bin_PROGRAMS = peacock
 
@@ -31,4 +33,6 @@
 	$(GNOMEUI_LIBS) \
 	$(GTKHTML_LIBS) \
 	$(LIBGLADE_LIBS) \
+	$(GDKPIXBUFF_LIBS) \
+	$(GDKPIXBUFFGNOME_LIBS) \
 	$(INTLLIBS) 
--- ./configure.in.org	Wed Oct  3 14:39:25 2001
+++ ./configure.in	Wed Jul 10 17:44:26 2002
@@ -67,6 +67,60 @@
 	AC_MSG_ERROR(LibGlade >= 0.16 needed for Peacock)
 fi
 
+
+dnl
+dnl gdk-pixbuf
+dnl
+AC_MSG_CHECKING(for gdk-pixbuf >= 0.17)
+if gdk-pixbuf-config --cflags > /dev/null 2>&1; then
+	vers=`gdk-pixbuf-config --version`
+	case $vers in
+		gdk-pixbuf-0.[1-16])
+			AC_MSG_RESULT(not found)
+			AC_MSG_ERROR(gdk-pixbuf > 0.17 needed for Peacock)
+			;;
+		*)
+			AC_MSG_RESULT(found)
+			GDKPIXBUFF_CFLAGS=`gdk-pixbuf-config --cflags`
+			GDKPIXBUFF_LIBS=`gdk-pixbuf-config --libs`
+			AC_SUBST(GDKPIXBUFF_CFLAGS)
+			AC_SUBST(GDKPIXBUFF_LIBS)
+			;;
+	esac
+else
+	AC_MSG_RESULT(not found)
+	AC_MSG_ERROR(gdk-pixbuf > 0.17 needed for Peacock)
+fi
+
+dnl
+dnl gdk-pixbuf-gnome
+dnl
+AC_MSG_CHECKING(for gdk-pixbuf-gnome >= 0.17)
+if gnome-config --libs gnomecanvaspixbuf > /dev/null 2>&1; then
+	vers=`gnome-config --version gnomecanvaspixbuf`
+	case $vers in
+		gnome-libs-1.[0-3])
+			AC_MSG_RESULT(not found)
+			AC_MSG_ERROR(gdk-pixbuf-gnome >= 0.17 needed for Peacock)
+			;;
+		gnome-libs-2)
+			AC_MSG_RESULT(not found)
+			AC_MSG_ERROR(GNOME2 is not supported yet...)
+			;;
+		*)
+			AC_MSG_RESULT(found)
+			GDKPIXBUFFGNOME_CFLAGS=`gnome-config --cflags gnomecanvaspixbuf`
+			GDKPIXBUFFGNOME_LIBS=`gnome-config --libs gnomecanvaspixbuf`
+			AC_SUBST(GDKPIXBUFFGNOME_CFLAGS)
+			AC_SUBST(GDKPIXBUFFGNOME_LIBS)
+			;;
+	esac
+else
+	AC_MSG_RESULT(not found)
+	AC_MSG_ERROR(gdk-pixbuf-gnome >= 0.17 needed for Peacock)
+fi
+
+
 dnl Add the languages which your application supports here.
 ALL_LINGUAS=""
 AM_GNU_GETTEXT
@@ -89,16 +143,14 @@
   AC_DEFINE_UNQUOTED(PACKAGE_GLADE_DIR, "${prefix}/${DATADIRNAME}/${PACKAGE}/glade")
 fi
 
-AC_OUTPUT([
-Makefile
-macros/Makefile
-src/Makefile
-src/dialogs/Makefile
-intl/Makefile
-po/Makefile.in
-doc/Makefile
-doc/C/Makefile
-pixmaps/Makefile
-peacock.desktop
-peacock.spec
-])
+AC_OUTPUT([Makefile
+	   macros/Makefile
+	   src/Makefile
+	   src/dialogs/Makefile
+	   intl/Makefile
+	   po/Makefile.in
+	   doc/Makefile
+	   doc/C/Makefile
+	   pixmaps/Makefile
+	   peacock.desktop
+	   peacock.spec])
