--- ./doc/C/Makefile.am.org	Wed Jul 10 16:34:04 2002
+++ ./doc/C/Makefile.am	Wed Jul 10 16:33:53 2002
@@ -33,10 +33,10 @@
 	-cp peacock.html $(distdir)
 
 install-data-local: peacock.html
-	$(mkinstalldirs) $(peacock_helpdir)/images
-	-for file in $(srcdir)/peacock/*.html $(srcdir)/peacock/*.css; do \
+	$(mkinstalldirs) $(DESTDIR)$(peacock_helpdir)/images
+	-for file in $(srcdir)/peacock/*.html $(DESTDIR)$(srcdir)/peacock/*.css; do \
 	basefile = `basename $$file`; \
-	$(INSTALL_DATA) $(srcdir)/$$file $(peacock_helpdir)/$$basefile; \
+	$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(peacock_helpdir)/$$basefile; \
 	done
 
 peacock.ps: peacock.sgml
